### ISSUE TITLE
Trigger Python event when `level_of_detail` mode finishes

### ISSUE THREAD

**Issue by jdugge** [bokeh/bokeh#3748]
It would be great if the Python script on a Bokeh server could be notified once a `level_of_detail` mode in the client (on zoom or pan) finishes, since this can be an appropriate point at which to generate updated data.

**Comment by bryevdv** [comment]
This seems reasonable, but I think we first need to build out a little capability for "events" that are not model attribute updates. E.g. we currently clumsily record button presses as an attribute state change, but buttons, and things like this should really be separate fire-and-forget type events back to the server, not model state updates. Thoughts @havocp ?

**Comment by havocp** [comment]
A simple version that might require only one new websocket message could be along these lines... 

```
class Model:
    def emit_event(self, name, args):
         """Broadcast an event with the given name and dict of args.
         
         The args dict cannot contain models, only plain old data.
         """
         pass
   def on_event(self, name, callback):
        """Register a callback to receive events with the given name and their args.
        """
```

On the Document and wire protocol level, there could be some kind of `ModelEventEvent` (better name possible), which would serialize the event name and args... so there's one new websocket message that has `name` and `args` in it.

**Comment by bryevdv** [comment]
@havocp that seems pretty reasonable, I've re-marked this as a feature.

**Comment by bryevdv** [comment]
This was closed by #5941
